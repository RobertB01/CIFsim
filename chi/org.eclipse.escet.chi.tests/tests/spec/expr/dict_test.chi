################################################################################
# Copyright (c) 2010, 2024 Contributors to the Eclipse Foundation
#
# See the NOTICE file(s) distributed with this work for additional
# information regarding copyright ownership.
#
# This program and the accompanying materials are made available under the terms
# of the MIT License which is available at https://opensource.org/licenses/MIT
#
# SPDX-License-Identifier: MIT
#################################################################################

model M():
    dict(int : string) dis;
    dis = {1:"a", 2:"b"+"b"};
    dis = dis + dis;
    writeln("%s[1]=%s", dis, dis[1]);
    dis[8] = "acht";
    writeln("%s+%s", dictkeys(dis), dictvalues(dis));
    dis = del(dis, 2);
    writeln("Dict after del(2): %s", dis);

    dis = {1:"a", 1:"b", 2:"a"}   # Fails on duplicate keys
end
