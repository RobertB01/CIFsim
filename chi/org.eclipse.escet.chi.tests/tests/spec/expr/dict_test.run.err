ERROR: Runtime error while executing "M".
at model "M" near line 23.
CAUSE: Literal dictionary has 1 duplicate key(s).
