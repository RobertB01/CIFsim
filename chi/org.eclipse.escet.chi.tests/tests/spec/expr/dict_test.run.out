{1 : a, 2 : bb}[1]=a
[1, 2, 8]+[a, bb, acht]
Dict after del(2): {1 : a, 8 : acht}
